### PR TITLE
Code optimization

### PR DESCRIPTION
**Pull request by zmberg**
### Ⅰ. Describe what this PR does
1. Optimize rollout state transition related code
2. e2e changed
3. changelog
4. github workflow e2e test

### CODE REVIEW

File: pkg/controller/rollout/batchrelease/batchrelease.go [hunk 1]
```diff
@@ -22,7 +22,8 @@ import rolloutv1alpha1 "github.com/openkruise/rollouts/api/v1alpha1"
 // but rather the ability to interact with the BatchRelease controller through the BatchRelease CRD to achieve a batch release
 type BatchRelease interface {
 	// Verify will create batchRelease or update batchRelease steps configuration
-	Verify(index int32) error
+	// isLatest(Bool) determines if the batchRelease configuration is the latest
```
Comment (furykerry) [inline pkg/controller/rollout/batchrelease/batchrelease.go:25]: what is the isLatest ? the return variables?

Comment (zmberg) [reply pkg/controller/rollout/batchrelease/batchrelease.go:25]: Mainly for the batch step adjustment scenario, this method will determine whether the configuration is consistent with the rollout step.

**Pull request merged**